argobots.spec

```
@@ -1,22 +1,27 @@
 Name: argobots
-Version: 1.0
-Release: 1%{?dist}
+Version: 1.1a1
+Release: 1%{?dist}.g7fd1987
 Summary: Lightweight, low-level threading and tasking framework
 Group: System Environment/Libraries
 License: UChicago Argonne, LLC -- Argobots License
 Url: http://www.argobots.org/
 #Source: https://api.github.com/repos/pmodels/$(NAME)/tarball/31703b1
 Source: https://github.com/pmodels/%{name}/releases/download/v%{version}/%{name}-%{version}.tar.gz
+# patch to gather all changes from v1.1a1 tag up to ULTs stack unwinding commit
+Patch0: from_v1.1a1_upto_7fd1987.patch
```

Our convention for naming these kind of patches is to use the `git` nomenclature to obtain it.

```suggestion
Patch0: v1.1a1...7fd1987.patch
```

---

> Looks like the patch isn't entirely clean:
> 
> ```
> [2021-01-21T19:53:26.180Z] dpkg-source: info: the patch has fuzz which is not allowed, or is malformed
> ```
> 

Well, I may not use the right way to create patch, I am using "git diff <tag_v1.1-a1_commit>..<unwind_commit>" when on Argobots master/main branch ...

> Also should update the packaging with:
> 
> ```
> $ cd ..
> $ git clone git@github.com:daos-stack/packaging.git
> $ git checkout origin/master
> $ cd -
> $ ../packaging/install
> ```

Sorry I am puzzled, where/when should I run/add this commands sequence ?? I believe the origin directory should be the daos/argobots repo local clone, right ? But then the "git checkout" and its following are very likely to fail... Also, why not make daos/packaging repo a submodule of daos/argobots ?

> Our convention for naming these kind of patches is to use the git nomenclature to obtain it.

Ok

---

Finally, instead of 
```
$ cd ..
$ git clone git@github.com:daos-stack/packaging.git
$ git checkout origin/master
$ cd -
$ ../packaging/install
```
I did
```
# CWD is my clone of daos/argobots set on my PR's branch
$ cd ..
$ git clone git@github.com:daos-stack/packaging.git
$ cd packaging
$ git checkout origin/master
$ cd ../<my clone of daos/argobots>
$ ../packaging/install
```
but then I have the updated copies from packaging to be git-added as new changed files, and also the "ln" command in ../packaging/install fails due to already existing files ./.git/hooks ...

Will try to push a new commit anyway and see how it goes.

---

Humm, even after I have pushed [528b538](https://github.com/daos-stack/argobots/pull/16/commits/528b538829566cdb9402a62851ec5e8042a66f15) , I still get "Packaging is out of date for argobots" email ...

---

> Well, I may not use the right way to create patch, I am using "git diff <tag_v1.1-a1_commit>..<unwind_commit>" when on Argobots master/main branch ...

Two things... Rather than using a commit of a tag, you can use the tag directly.  I'm not entirely sure why, but patches come out cleaner when you use `...` for the diff instead of `..`.  So instead: `git diff v1.1a1...<unwind_commit>`.

> Sorry I am puzzled, where/when should I run/add this commands sequence ??

Before you commit changes.  They need to be included in the commit.

> I believe the origin directory should be the daos/argobots repo local clone, right ?

Correct.

> But then the "git checkout" and its following are very likely to fail... Also, why not make daos/packaging repo a submodule of daos/argobots ?

Because Jenkins doesn't check out submodules when it starts it's job and it becomes a catch-22, in that the `packaging/` submodule needs to be present before anything can be done.  Believe me, I would love nothing better than to make it a submodule, pointing at `master` (rather than a commit, even).

---

> ```
>...
> $ cd packaging
> $ git checkout origin/master
> ...
> ```

Good catch, but in fact, after the clone `packaging/` should already be on `origin/master` as that is the default branch for that repo.

> 
> but then I have the updated copies from packaging to be git-added as new changed files, and also the "ln" command in ../packaging/install fails due to already existing files ./.git/hooks ...

Yes, you can ignore that if you already have some .git/hooks.  We need to find a better, general way to handle githooks.  It's really quite disappointing how git doesn't actually commit them to a repo.

---

> Humm, even after I have pushed [528b538](https://github.com/daos-stack/argobots/pull/16/commits/528b538829566cdb9402a62851ec5e8042a66f15) , I still get "Packaging is out of date for argobots" email ...
> May be I need to "rm .git/hook/*" before running "../packaging/install" ??

No.  As I said, you can ignore that.  Your problem is that you are missing daos-stack/packaging@2cde278.  You must have not updated your packaging repo since you last cloned it.  Update it and run `../packaging/install` again to get that latest update.  That one fixes at least one other problem in your CI run.